ssl/ssl_locl.h

```diff
@@ -668,6 +668,8 @@ typedef struct raw_extension_st {
     int parsed;
     /* The type of this extension, i.e. a TLSEXT_TYPE_* value */
     unsigned int type;
+    /* The (0-based) index of this extension in its array, if present */
+    size_t idx;
```

Comment:
The ClientHello is the only place we currently use this field for, yes.  but we store it everywhere we parse extensions.  (IIRC the language was a bit stilted due to line-length constraints.)  I'll reword and rename.